regular_expressions_intermediate/regular_expressions_intermediate.md

```diff
@@ -0,0 +1,540 @@
+</div>
+***
+
+### Non-Capturing Groups and the Logical "Or"
+
+Sometimes you want to define two different pattern options at a certain point in a complex string.  For example, let's consider a mouse model study of the immune system's interaction with insulin.  
+
+Your wild type subjects are given identifiers that include `WT` (wild type), followed by a hyphen, and a subject identifier of usually five (but possibly 6) digits.  You additionally have PBMC humanized mice, in addition to wild type mice, and their identifiers begin with `PBMC`, followed by a hyphen, followed by 5-6 digits.
+
+There's a handy way to indicate a logical "or" in regex, using the vertical pipe (`|`).  Tokens on either side of a `|` character are alternative values for the given part of a pattern. 
+
+For example, `8|5` indicates that either an 8 or a 5 appears at a given location. `banana|B` indicates that the word "banana" or the abbreviation "B" appears.  And `NA|\d{4}|unknown` indicates that one of "NA", a four digit number, or "unknown" should appear.
+
+So, `WT|PBMC` is a good start.  However, that's not the only thing in our pattern.  We have to add a hyphen (`\-`) and then 5-6 digits (`\d{5,6}`).  
+
+Would `WT|PBMC\-\d{5,6}` work?  No!  That would be interpreted as one of the following:
+
+* 'WT' (by itself) **or**
+* 'PBMC' followed by a hyphen and 5-6 digits.
+
+That's not what we mean!  And we can't use a space to our regex to syntactically separate our "or" clause from the rest of the regular expression. After all, a space is interpreted literally, as a space that belongs in the pattern.  So, what can we do? We can use parentheses for grouping. 
+
+`(WT|PBMC)\-\d{5,6}` would work, and would be interpreted as:
+
+* "WT" or "PBMC" (and capture whichever it is)
+* Then a hyphen
+* Then 5-6 digits
+
+[That expression will work as valid regex](https://regex101.com/r/ZHs5iI/1) that will match with your various mouse subject ids.  However, you really don't need to capture that group, you're just using the parentheses to enclose a group that could go one of two different ways.  If you prefer, you can indicate that it's a non-capturing group by adding `?:` after the opening parenthesis, like this:
+
+`(?:WT|PBMC)\-\d{5,6}`
+
+[This will also work](https://regex101.com/r/ko7ySn/1) as a matching regular expression, but won't capture the group in question.
+
+### Quiz 3: Non-Capturing Groups and the Logical "Or"
+
+You're looking through lab notes written by a colleague who uses codes and abbreviations liberally.  This investigator is known to write their lowercase initials ("sjp") and a single space followed by either "help" (always in lowercase) or "911" in notes to highlight important potential pitfalls.  Using the logical "or" and using a non-capturing group (remember: the syntax is `(?:)`), write a regular expression that would allow you to capture any of the "sjp help" or "sjp 911" references.
+
+Put that regular expression in the box below:
+
+[[sjp (?:help|911)]]
```

Comment:
This answer is rendering as a dropdown menu box using the pipe as a line divider. One way to hack this might be to make this a multiple choice question with a bunch of options, although that defeats the "you have to type it out yourself to learn" goal here.... maybe an escape character will help?

Reply:
Fixed it in a new commit I think!